examples/src/main/java/com/mapbox/navigation/examples/NavigationApplication.kt

```diff
@@ -58,5 +60,6 @@ class NavigationApplication : MultiDexApplication() {
         }
 
         Mapbox.getInstance(applicationContext, mapboxAccessToken)
+        MapboxNativeNavigatorImpl.create(HandheldProfile())
```

Comment:
this makes me feel like we could have a MapboxNavigation.create(..)

some things would belong there

Reply:
More information. What's causing one of the crashes is our examples settings changes the settings of MapboxNavigation before it is constructed. [Specifically here](https://github.com/mapbox/mapbox-navigation-android/blob/25e65edc73272fefb0afa199356e4689c8347cbc/examples/src/main/java/com/mapbox/navigation/examples/MainActivity.kt#L102-L104) 

Fixing the crash for now, long term fix would not use an internal interface. Will also need to delete the usage of `MapboxNativeNavigatorImpl.toggleHistory(`